src/types.ts

```diff
@@ -48,15 +51,36 @@ export type RequestPayloadOf<
   Route extends keyof Endpoints,
 > = Endpoints[Route]['Request'] & PathParamsOf<Route>;
 
-type QueryOptions<Response> =
-  | UseQueryOptions<Response, unknown>
-  | UseInfiniteQueryOptions<Response, unknown>;
+type RestrictedUseQueryOptions<Response> = Omit<
+  UseQueryOptions<Response, unknown>,
+  'queryKey' | 'queryFn'
+> & {
+  axios?: AxiosRequestConfig;
+};
 
-type RestrictedUseQueryOptions<
-  Response,
-  TQueryOptions extends QueryOptions<Response> = UseQueryOptions<Response>,
-> = Omit<TQueryOptions, 'queryKey' | 'queryFn'> & {
+type RestrictedUseInfiniteQueryOptions<Response, Request> = Omit<
```

Comment:
nit: the naming is pretty verbose for this type, since we do expose it in the library. Maybe `InfiniteQueryOptions` and `InfiniteQueryResult` is best?

Reply:
I don't think this type is explicitly exposed -- personally, I like the verbosity either way. React Query's types can be verbose, and it feels like we're sticking with their pattern